### PR TITLE
fix(drive): mishandling internal errors as validation ones

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

Rework drive abci endpoints to not mishandle internal erros as validation ones

## How Has This Been Tested?

- CI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by markin-io** [comment]
@shumkov do you think we want to `Err` here instead of returning Internal Error?
https://github.com/dashpay/platform/blob/v0.25-dev/packages/rs-drive-abci/src/abci/handlers.rs#L620

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 1]
```diff
@@ -657,27 +773,32 @@ impl<C> Platform<C> {
...
                 let contract = check_validation_result_with_data!(contract.ok_or(
-                    QueryError::Query(QuerySyntaxError::DataContractNotFound(
-                        "contract not found when querying from value with contract info",
-                    ))
+                    QueryError::NotFound(format!("data contract {} not found", contract_id))
                 ));
+
                 let contract_ref = &contract.contract;
+
                 let document_type = check_validation_result_with_data!(
```
Comment (shumkov) [inline packages/rs-drive-abci/src/query/v0/mod.rs:802]: it should be an invalid argument

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 2]
```diff
@@ -898,24 +1038,32 @@ impl<C> Platform<C> {
...
                         .drive
                         .fetch_full_identities_by_unique_public_key_hashes(
                             public_key_hashes.as_slice(),
                             None,
                             platform_version,
-                        ));
-                    let identities = check_validation_result_with_data!(identities
+                        )?;
+
+                    if identities.is_empty() {
```
Comment (shumkov) [inline packages/rs-drive-abci/src/query/v0/mod.rs:1049]: We requesting multiple identities, we respond with an empty set if non of them found

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 3]
```diff
@@ -611,29 +718,38 @@ impl<C> Platform<C> {
...
+                    let contracts = self.drive.fetch_contract_with_history(
+                        contract_id.to_buffer(),
+                        None,
+                        start_at_ms,
+                        limit,
+                        offset,
+                        platform_version,
+                    )?;
+
+                    if contracts.is_empty() {
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs:730]: Isn't it similar to fetchDataContract? We have a singular contract_id and if we have an empty history, we signalize that contract is kinda not found?

File: packages/rs-drive-abci/src/error/query.rs [hunk 4]
```diff
@@ -13,8 +13,9 @@ pub enum QueryError {
     /// Proof Error
     #[error("proof error: {0}")]
     Proof(#[from] ProofError),
+
     /// Syntax Error
-    #[error("query syntax error: {0}")]
+    #[error("document query syntax error: {0}")]
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/rs-drive-abci/src/error/query.rs]: Fixed

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 5]
```diff
@@ -232,19 +260,29 @@ impl<C> Platform<C> {
...
-                            None,
-                            &platform_version.drive
-                        ));
+                    let proof = self.drive.prove_identity_balance(
+                        identity_id.into_buffer(),
+                        None,
+                        &platform_version.drive,
+                    )?;
+
+                    if proof.is_empty() {
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs]: Fixed

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 6]
```diff
@@ -258,35 +296,55 @@ impl<C> Platform<C> {
...
-                            None,
-                            &platform_version.drive
-                        ));
+                    let proof = self.drive.prove_identity_balance_and_revision(
+                        identity_id.into_buffer(),
+                        None,
+                        &platform_version.drive,
+                    )?;
+
+                    if proof.is_empty() {
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs]: Fixed

File: packages/rs-drive-abci/src/query/v0/mod.rs [hunk 7]
```diff
@@ -557,47 +657,54 @@ impl<C> Platform<C> {
...
+                    let proof = self.drive.prove_contract_history(
+                        contract_id.to_buffer(),
+                        None,
+                        start_at_ms,
+                        limit,
+                        offset,
+                        platform_version,
+                    )?;
+
+                    if proof.is_empty() {
```
[the remark this answers is not in the document]
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs]: Fixed

File: packages/rs-drive-abci/src/query/v0/mod.rs (hunk 1 above)
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs]: Fixed

File: packages/rs-drive-abci/src/query/v0/mod.rs (hunk 2 above)
Comment (markin-io) [reply packages/rs-drive-abci/src/query/v0/mod.rs]: Fixed

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]
Please fix clippy warnings.

**Review by shumkov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**